src/github3/checks.py

```diff
@@ -0,0 +1,416 @@
+
+    refresh = to_app
+
+
+class CheckSuite(models.GitHubCore):
+    """The :class:`CheckSuite <CheckSuite>` object.
+
+    .. versionadded:: 1.3.0
+
+    Please see GitHub's `CheckSuite Documentation`_ for more information.
+
+    .. attribute:: status
+
+        The status of the Check Suite
+
+    .. attribute:: conclusion
+
+        The highest priority check run conclusion. If it has not completed this
+        will be None
+
+    .. attribute:: head_sha
+
+        The sha of the commit at the head of the branch the check was run
+        against (the source of the pull request)
+
+    .. attribute:: head_branch
+
+        The branch checked
+
+    .. attribute:: before
+
+        The sha of the pull request target branch at the time of the checks
+
+    .. attribute:: after
+
+        The sha of the target branch after the change is applied
+
+    .. attribute:: repository
+
+        A representation of the repository the suite belongs to as
+        :class:`~github3.repos.repo.ShortRepository`.
+
+    .. attribute:: origional_pull_requests
+
+        A list of representations of the pull requests the suite belongs to as
+        :class:`~github3.checks.CheckPullRequest`.
+
+        .. note::
+
+            This may be empty.
+
+    .. attribute:: id
+
+        The unique GitHub assigned numerical id of this check suite.
+
+    .. attribute:: app
+
+        A :class:`~github3.checks.CheckApp` representing the App
+        this suite belongs to.
+
+    .. CheckSuite Documentation:
+        http://developer.github.com/v3/checks/suites/
+    """
+
+    class_name = 'CheckSuite'
+    CUSTOM_HEADERS = {
+        'Accept': 'application/vnd.github.antiope-preview+json'
+    }
+
+    def _update_attributes(self, suite):
+        # Import here, because a toplevel import causes an import loop
+        from . import repos
+        self._api = suite['url']
+        self.status = suite['status']
+        self.conclusion = suite['conclusion']
+        self.head_branch = suite['head_branch']
+        self.head_sha = suite['head_sha']
+        self.before = suite['before']
+        self.after = suite['after']
+        prs = suite.get('pull_requests', [])
+        self.origional_pull_requests = [
+            CheckPullRequest(p, self) for p in prs
+        ]
+        self.repository = repos.ShortRepository(suite['repository'], self)
+        self.id = suite['id']
+        self.app = CheckApp(suite['app'], self)
+
+    def _repr(self):
+        return '<{s.class_name} [{s.id}:{s.status}]>'.format(s=self)
+
+    @decorators.requires_app_installation_auth
+    def rerequest(self):
+        """Rerequest the check suite.
+
+        :returns:
+            True if successful, False otherwise
+        :rtype:
+            bool
+        """
+        url = self._build_url('rerequest', base_url=self._api)
+        return self._boolean(self._post(
+            url, headers=CheckSuite.CUSTOM_HEADERS), 201, 404)
+
+    @decorators.requires_app_installation_auth
```

Comment:
I think some of the methods don't require app installation auth. Seems to be only required for write access.

From the docs:
> Write permission for the Checks API is only available to GitHub Apps. OAuth Apps and authenticated users can view check runs and check suites, but they are not able to create them.

Reply:
That's a good point.